openslides/agenda/static/js/agenda/base.js

```diff
@@ -23,21 +23,26 @@ angular.module('OpenSlidesApp.agenda', ['OpenSlidesApp.users'])
 
 .factory('AgendaUpdate',[
     'Agenda',
-    function(Agenda) {
+    'operator',
+    function(Agenda, operator) {
         return {
             saveChanges: function (item_id, changes) {
-                Agenda.find(item_id).then(function(item) {
-                    var something = false;
-                    _.each(changes, function(change) {
-                        if (change.value !== item[change.key]) {
-                            item[change.key] = change.value;
-                            something = true;
+                // change agenda item only if user has the permission to do that
+                if (operator.hasPerms('agenda.can_manage agenda.can_see_hidden_items')) {
+                    Agenda.find(item_id).then(function (item) {
```

Comment:
Ah this is bad. The problem is that we wait for the response of the (e. g. motion) create request and then want to fire a new agenda update request but have to wait for the server to create the agenda item.

Suggestion: We refactor the create requests and let the server do the changes. Then we remove this code here completely.

Please open an issue, if you think, this is good.

Hint: At the moment we have code also to update agenda items on (e. g. motion) update requests but we do not use it because we removed the missleading form fields (flag parent item) or we should remove this form fields (hidden item flag).